reporter.cpp

```diff
@@ -0,0 +1,236 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <iostream>
+#include <nlohmann/json.hpp>
+#include <regex>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+// for convenience
+using json = nlohmann::json;
+using namespace std;
+
+typedef struct Value {
+  string value;
```

Comment:
Consdier using `JSONValue` or something else to avoid confusion with LLVM `Value`